ops/gmpctl/main.go

```diff
@@ -0,0 +1,233 @@
+)
+
+var (
+	cfgPath        = flag.String("c", ".gmpctl.default.yaml", "Path to the configuration file. See config.go#Config for the structure.")
+	verbose        = flag.Bool("v", false, "Enabled verbose, debug output (e.g. logging os.Exec commands)")
+	gitPreferHTTPS = flag.Bool("git.prefer-https", false, "If true, uses HTTPS protocol instead of git for remote URLs. ")
+)
+
+type Config struct {
+	// Directory for the gmpctl work, notably for project clones and git worktrees.
+	Directory string `yaml:"dir"`
+}
+
+func loadConfig() (ret *Config, _ error) {
+	b, err := os.ReadFile(*cfgPath)
+	if err != nil {
+		return nil, err
+	}
+	if err := yaml.Unmarshal(b, &ret); err != nil {
+		return nil, err
+	}
+	ret.Directory, err = filepath.Abs(ret.Directory)
+	return ret, err
+}
+
+func main() {
+	flag.Usage = func() {
+		fmt.Println("Usage: gmpctl [COMMAND] [FLAGS]")
+		flag.PrintDefaults()
+		fmt.Print("\n--- Commands ---\n")
+		fmt.Print("[release] ")
+		releaseFlags.Usage()
+		fmt.Println()
+		fmt.Print("[vulnfix] ")
+		vulnfixFlags.Usage()
+	}
+
+	// Help.
+	for _, cmd := range os.Args[1:] {
+		if cmd == "-h" || cmd == "--help" {
+			flag.Usage()
+			os.Exit(0)
+		}
+	}
+
+	flag.Parse()
+	if flag.NArg() == 0 {
+		errf("expected at least one argument, got none")
+		flag.Usage()
+		os.Exit(1)
+	}
+
+	var (
+		err error
+		cmd = flag.Arg(0)
+	)
+	switch cmd {
+
+	case "release":
+		err = release()
+	case "vulnfix":
+		err = vulnfix()
+	default:
+		errf("Unknown command: %q", cmd)
+		flag.Usage()
+		os.Exit(1)
+	}
+
+	if err != nil {
+		errf("Command %q failed: %v", cmd, err)
+		os.Exit(1)
+	}
+	successf("Command %q succeded", cmd)
+	os.Exit(0)
+}
+
+type cmdOpts struct {
+	// Dir configures the directory that command will be run from.
+	Dir string
+
+	// Envs configures additional OS environments.
+	Envs []string
+	// HideOutputs disables stdout and stderr streaming.
+	// This is useful for ~porcelain like commands that are meant to pass state via
+	// stdout.
+	HideOutputs bool
+}
+
+// libScriptFile is contains useful shell functions.
+// Sometimes it's just easier to hack something in bash before porting to Go.
+// TODO(bwplotka): go embed this for portability?
+const libScriptFile = "lib.sh"
+
+// getFromLibFunction runs certain function from libScript that's expected to pass a return
+// parameter via stdout.
+func getFromLibFunction(dir string, envs []string, function string, args ...string) (string, error) {
+	curr, err := filepath.Abs("") // Hacky. TODO(bwplotka): Improve dir management.
+	if err != nil {
+		return "", err
+	}
+	libScript := filepath.Join(curr, libScriptFile)
+
+	envs = append(envs,
+		fmt.Sprintf("SCRIPT_DIR=%v", curr),
+	)
+	return runCommand(
+		&cmdOpts{Dir: dir, Envs: envs, HideOutputs: true},
+		"bash", "-c", fmt.Sprintf(". %v && %v %v", libScript, function, strings.Join(args, " ")),
+	)
+}
+
+// runLibFunction runs certain function from libScript that is not expected
+// to pass any return parameters.
```

Comment:
Do we have to jump back and forth between Go code and Bash scripts? Why not implement these bash functions in Go?

Reply:
Because of time constraints, because of how quickly I could experiment with bash. So it is an iteration. I'm sorry but I won't be able to port everything now. 

I would love to prioritize use and enablement of automation for our release process for the team. 

Is the bash use a blocker for you? We had consensus about it already (see our docs).